Dockerizing the pythondotorg codebase: (#2130)
proevo/pythondotorg: docs/source/install.md

```
git clone git@github.com:YOUR-USERNAME/pythondotorg.git
```

Add a [remote](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/configuring-a-remote-for-a-fork) and [sync](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork) regularly to stay current with the repository. 

```
git remote add upstream https://github.com/python/pythondotorg
git checkout main
git fetch upstream 
git merge upstream/main 
```

Installing Docker
-----------------

Install [Docker Engine](https://docs.docker.com/engine/install/) 

```{note}
The best experience for building Pythondotorg on Windows is to use the [Windows Subsystem for Linux](https://docs.microsoft.com/en-us/windows/wsl/)(WSL) in combination with both [Docker for Windows](https://docs.docker.com/desktop/install/windows-install/) and [Docker for Linux](https://docs.docker.com/engine/install/).
```

Verify that the Docker installation is successful by running: `docker -v`

Running pythondotorg locally 
----------------------------
Once you have Docker and Docker Compose installed, run:

```
make serve
```

> This will pull down all the required docker containers, build Pythondotorg and establish all of the neccessary services. 

Reviewer: ```suggestion
This will pull down all the required docker containers, build the environment for pythondotorg, run migrations, load development fixtures, and start all of the necessary services. 
```

> ```

Reviewer: ````suggestion
Once complete, you will see the following in your terminal output:

```
````

web_1       | Starting development server at http://0.0.0.0:8000/
web_1       | Quit the server with CONTROL-C.
``` 

You can view these results in your local web browser at: `http://localhost:8000`

To reset your local environment, run:

```
make clean
```

> To apply new migrations, run: 

Reviewer: ```suggestion
To apply migrations, run: 
```

``` 
make migrate
```

> To create new migrations, run: 

Reviewer: ```suggestion
To generate new migrations, run: 
```

```
make migrations
```

> 

Reviewer: ````suggestion

You can also run arbitrary Django management commands via:

```
make manage <NAME_OF_COMMAND>
```

This is a simple wrapper around running `python manage.py` in the container, all arguments passed to `make manage` will be passed through.

````

```
git clone git@github.com:YOUR-USERNAME/pythondotorg.git
```

Add a [remote](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/configuring-a-remote-for-a-fork) and [sync](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork) regularly to stay current with the repository. 

```
git remote add upstream https://github.com/python/pythondotorg
git checkout main
git fetch upstream 
git merge upstream/main 
```

Installing Docker
-----------------

Install [Docker Engine](https://docs.docker.com/engine/install/) 

```{note}
The best experience for building Pythondotorg on Windows is to use the [Windows Subsystem for Linux](https://docs.microsoft.com/en-us/windows/wsl/)(WSL) in combination with both [Docker for Windows](https://docs.docker.com/desktop/install/windows-install/) and [Docker for Linux](https://docs.docker.com/engine/install/).
```

Verify that the Docker installation is successful by running: `docker -v`

> Building the Pythondotorg Container

Reviewer: I think this could be renamed "Running pythondotorg locally".

-----------------------------------
Once you have Docker and Docker Compose installed, run:

```
make serve
```

> This will pull down all the required docker containers, build Pythondotorg and establish all of the neccessary services. 

Reviewer: It might be good to show the expected output ala:

```
web_1       | Starting development server at http://0.0.0.0:8000/
web_1       | Quit the server with CONTROL-C.
```

And also instruct people to view the result in their local web browser at `http://localhost:8000`

To reset your local environment, run:

```
make clean
```

> To create or apply new migrations, run: 

Reviewer: This will only apply migrations, we need a separate note for creating new ones `make migrations`.
